coopaname_custom_contract/models/hr_contract.py

```diff
@@ -0,0 +1,223 @@
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class Contract(models.Model):
+    _inherit = "hr.contract"
+
+    @api.multi  # noqa
+    def create_document(self):
+        self.ensure_one()
+        self.check_existing_document()
+        if (
+            self.type_id
+            == self.env.ref("hr_cae_contract.hr_contract_type_bonus")
+            and self.echelon == "amendment"
+        ):
+            return self.create_report_action(
+                "coopaname_custom_contract.report_hr_cae_contract_bonus",
+                "Bonus",
+            )
```

Comment:
Improvement suggestion, replace the if ... elif ... else by a dictionnary. Something like

```python
REPORT_NAMES = {
  (self.env.ref("hr_cae_contract.hr_contract_type_bonus").id, "amendment"): "coopaname_custom_contract.report_hr_cae_contract_bonus",
...
}
report_name = REPORT_NAMES.get((self.type_id.id, self.echelon), False)
if report_name:
    return self.create_report_action(report_name)
else: 
    raise
```

I can elaborate tomorrow but now it's late.